agent/consul/rpc.go

```diff
@@ -949,3 +952,58 @@ func (s *Server) consistentRead() error {
 
 	return structs.ErrNotReadyForConsistentReads
 }
+
+type rpcMetricsInterceptor struct {
+	// TODO: accept a smaller interface
+	srv *Server
+}
+
+func (r rpcMetricsInterceptor) ServiceCall(fn func(), serviceMethod string, argv, replyv reflect.Value) {
+	start := time.Now()
+	fn()
+
+	failed := "false"
+	if _, ok := replyv.Interface().(error); ok {
```

Comment:
i'm just leaving this as a note to myself:

`replyv` here might actually come out as a `*net.rpc.Reply` (ptr) ; so `.(error)` type conversion may panic/ not work as intended.